packages/desktop-client/src/components/util/PercentInput.tsx

```diff
@@ -0,0 +1,125 @@
+import React, {
+  type Ref,
+  useRef,
+  useState,
+  useEffect,
+  type FocusEventHandler,
+  type FocusEvent,
+} from 'react';
+
+import { evalArithmetic } from 'loot-core/src/shared/arithmetic';
+
+import { useMergedRefs } from '../../hooks/useMergedRefs';
+import { type CSSProperties } from '../../style';
+import { Input } from '../common/Input';
+import { useFormat } from '../spreadsheet/useFormat';
+
+type PercentInputProps = {
+  id?: string;
+  inputRef?: Ref<HTMLInputElement>;
+  value: number;
+  defaultValue?: number;
+  onChangeValue?: (value: string) => void;
+  onFocus?: FocusEventHandler<HTMLInputElement>;
+  onBlur?: FocusEventHandler<HTMLInputElement>;
+  onUpdatePercent?: (percent: number) => void;
```

Comment:
Nit: make same order as below.
```suggestion
  onFocus?: FocusEventHandler<HTMLInputElement>;
  onBlur?: FocusEventHandler<HTMLInputElement>;
  onChangeValue?: (value: string) => void;
  onUpdatePercent?: (percent: number) => void;
```